app/scripts/services/dimItemInfoService.factory.js

```diff
@@ -0,0 +1,64 @@
+(function() {
+  'use strict';
+
+  angular.module('dimApp')
+    .factory('dimItemInfoService', ItemInfoService);
+
+  ItemInfoService.$inject = ['dimPlatformService', 'SyncService'];
+
+  /**
+   * The item info service maintains a map of extra, DIM-specific, synced data about items (per platform).
+   * These info objects have a save method on them that can be used to persist any changes to their properties.
+   */
+  function ItemInfoService(dimPlatformService, SyncService) {
+    // Returns a function that, when given a platform, returns the item info source for that platform
+    return function(platform) {
+      return SyncService.get().then(function(data) {
+        const key = 'dimItemInfo-' + platform.type;
+        const infos = data[key] || {};
+
+        return {
+          infoForItem: function(hash, id) {
```

Comment:
I get what you were saying now. Thanks for implementing it this way. Makes sense.